cpp/include/raft/distance/detail/fused_l2_nn.cuh

```diff
@@ -299,58 +322,73 @@ void fusedL2NNImpl(OutT* min,
     RAFT_CUDA_TRY(cudaGetLastError());
   }
 
-  constexpr size_t shmemSize = P::SmemSize + ((P::Mblk + P::Nblk) * sizeof(DataT));
-  if (sqrt) {
-    auto fusedL2NNSqrt = fusedL2NNkernel<DataT,
-                                         OutT,
-                                         IdxT,
-                                         true,
-                                         P,
-                                         ReduceOpT,
-                                         KVPReduceOpT,
-                                         decltype(core_lambda),
-                                         raft::identity_op>;
-    dim3 grid          = launchConfigGenerator<P>(m, n, shmemSize, fusedL2NNSqrt);
-
-    fusedL2NNSqrt<<<grid, blk, shmemSize, stream>>>(min,
-                                                    x,
-                                                    y,
-                                                    xn,
-                                                    yn,
-                                                    m,
-                                                    n,
-                                                    k,
-                                                    maxVal,
-                                                    workspace,
-                                                    redOp,
-                                                    pairRedOp,
-                                                    core_lambda,
-                                                    raft::identity_op{});
+  const auto deviceVersion = getComputeCapability();
+
+  if (deviceVersion.first >= 8) {
+    using L2Op                  = L2ExpandedOp<DataT, DataT>;
```

Comment:
PR #1142 moved things around a bit. The L2ExpandedOp can be obtained with:
```
#include <raft/distance/detail/distance_ops/l2_exp.cuh>

      using L2Op = ops::l2_exp_cutlass_op<DataT, AccT>;
      L2Op l2_op(perform_sqrt);
```
See: https://github.com/rapidsai/raft/blob/b12fc82be4c06cd157925ef14411fa2ee9cf0d07/cpp/include/raft/distance/detail/distance.cuh#L533